# Desafio da Disciplina Visualização de Dados

## Enunciado

Qual é a diferente entre alunos que têm os pais casados e separados que vivem ou não na mesma casa e que conseguiram pagar os estudos e têm interesse em fazer um curso superior ?
- Vamos agrupar por sexo.
- Vamos agrupar por pais que têm ensino superior.

Após isso, vamos apresentar quais diferença entre os grupos e identificar quais pontos podemos melhorar para ter notas superiores em Português e Matemática.

## Dicionário de Dados

### Student Performance
Os dados podem ser baixados neste [link](https://archive.ics.uci.edu/dataset/320/student+performance).

#### Atributos para os conjuntos de dados student-mat.csv (curso de matemática) e student-por.csv (curso de língua portuguesa):

1. school - escola do aluno (binário: 'GP' - Gabriel Pereira ou 'MS' - Mousinho da Silveira)
2. sex - sexo do aluno (binário: 'F' - feminino ou 'M' - masculino)
3. age - idade do aluno (numérico: de 15 a 22)
4. address – tipo de endereço residencial do aluno (binário: ‘U’ – urbano ou ‘R’ – rural)
5. famsize - tamanho da família (binário: 'LE3' - menor ou igual a 3 ou 'GT3' - maior que 3)
6. Pstatus - situação de coabitação dos pais (binário: 'T' - morando juntos ou 'A' - separados)
7. Medu - escolaridade da mãe (numérico: 0 - nenhuma, 1 - ensino fundamental (4º ano), 2 - 5º ao 9º ano, 3 - ensino médio ou 4 - ensino superior)
8. Fedu - escolaridade do pai (numérico: 0 - nenhuma, 1 - ensino fundamental (4º ano), 2 - 5º ao 9º ano, 3 - ensino médio ou 4 - ensino superior)
9. Mjob - trabalho da mãe (nominal: 'professor', 'cuidados de saúde', 'serviços' civis (por exemplo, administrativo ou policial), 'em_casa' ou 'outros')
10. Fjob - trabalho do pai (nominal: 'professor', 'cuidados de saúde', 'serviços' civis (por exemplo, administrativo ou policial), 'em_casa' ou 'outros')
11. reason - motivo para escolher esta escola (nominal: perto de ‘casa’, ‘reputação’ da escola, preferência de ‘curso’ ou ‘outro’)
12. guardian - tutor do aluno (nominal: 'mãe', 'pai' ou 'outro')
13. traveltime - tempo de viagem da casa até a escola (numérico: 1 - <15 min., 2 - 15 a 30 min., 3 - 30 min. a 1 hora, ou 4 - >1 hora)
14. studytime - tempo de estudo semanal (numérico: 1 - <2 horas, 2 - 2 a 5 horas, 3 - 5 a 10 horas ou 4 - >10 horas)
15. failures - número de falhas de classe anteriores (numérico: n se 1<=n<3, caso contrário 4)
16. schoolup - apoio educativo extra (binário: sim ou não)
17.  famsup – apoio educacional familiar (binário: sim ou não)
18. paid - extras pagas dentro da disciplina do curso (Matemática ou Português) (binário: sim ou não)
19. activities - atividades extracurriculares (binário: sim ou não)
20. nursery - frequentou a creche (binário: sim ou não)
21. higher - quer cursar ensino superior (binário: sim ou não)
22. internet - Acesso à internet em casa (binário: sim ou não)
23. romantic - com relacionamento amoroso (binário: sim ou não)
24. famrel - qualidade das relações familiares (numérico: de 1 - péssimo a 5 - excelente)
25. freetime - tempo livre depois da escola (numérico: de 1 - muito baixo a 5 - muito alto)
26. goout - sair com amigos (numérico: de 1 - muito baixo a 5 - muito alto)
27. Dalc - consumo de álcool durante a jornada de trabalho (numérico: de 1 - muito baixo a 5 - muito alto)
28. Walc - consumo de álcool no final de semana (numérico: de 1 - muito baixo a 5 - muito alto)
29. health – estado de saúde atual (numérico: de 1 – muito ruim a 5 – muito bom)
30. absences - número de faltas escolares (numérico: de 0 a 93)

#### Essas notas estão relacionadas com a disciplina do curso, Matemática ou Português:

31. G1 - nota do primeiro período (numérico: de 0 a 20)

31. G2 - nota do segundo período (numérico: de 0 a 20)

32. G3 - nota final (numérico: de 0 a 20, meta de saída)


## Resolução

In [1]:
import pandas as pd

In [2]:
from google.colab import files
uploaded = files.upload()

Saving student-mat.csv to student-mat.csv
Saving student-por.csv to student-por.csv


### Leitura e junção dos dados

In [4]:
df_pt = pd.read_csv("student-por.csv", sep=";")
df_mat = pd.read_csv("student-mat.csv", sep=";")

##### Português

In [5]:
df_pt.head()

school sex  age address famsize Pstatus  Medu  Fedu     Mjob      Fjob  ...  \
0     GP   F   18       U     GT3       A     4     4  at_home   teacher  ...   
1     GP   F   17       U     GT3       T     1     1  at_home     other  ...   
2     GP   F   15       U     LE3       T     1     1  at_home     other  ...   
3     GP   F   15       U     GT3       T     4     2   health  services  ...   
4     GP   F   16       U     GT3       T     3     3    other     other  ...   

  famrel freetime  goout  Dalc  Walc health absences  G1  G2  G3  
0      4        3      4     1     1      3        4   0  11  11  
1      5        3      3     1     1      3        2   9  11  11  
2      4        3      2     2     3      3        6  12  13  12  
3      3        2      2     1     1      5        0  14  14  14  
4      4        3      2     1     2      5        0  11  13  13  

[5 rows x 33 columns]

In [6]:
df_pt.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 649 entries, 0 to 648
Data columns (total 33 columns):
 #   Column      Non-Null Count  Dtype 
---  ------      --------------  ----- 
 0   school      649 non-null    object
 1   sex         649 non-null    object
 2   age         649 non-null    int64 
 3   address     649 non-null    object
 4   famsize     649 non-null    object
 5   Pstatus     649 non-null    object
 6   Medu        649 non-null    int64 
 7   Fedu        649 non-null    int64 
 8   Mjob        649 non-null    object
 9   Fjob        649 non-null    object
 10  reason      649 non-null    object
 11  guardian    649 non-null    object
 12  traveltime  649 non-null    int64 
 13  studytime   649 non-null    int64 
 14  failures    649 non-null    int64 
 15  schoolsup   649 non-null    object
 16  famsup      649 non-null    object
 17  paid        649 non-null    object
 18  activities  649 non-null    object
 19  nursery     649 non-null    object
 20  higher    

In [7]:
df_mat.head()

school sex  age address famsize Pstatus  Medu  Fedu     Mjob      Fjob  ...  \
0     GP   F   18       U     GT3       A     4     4  at_home   teacher  ...   
1     GP   F   17       U     GT3       T     1     1  at_home     other  ...   
2     GP   F   15       U     LE3       T     1     1  at_home     other  ...   
3     GP   F   15       U     GT3       T     4     2   health  services  ...   
4     GP   F   16       U     GT3       T     3     3    other     other  ...   

  famrel freetime  goout  Dalc  Walc health absences  G1  G2  G3  
0      4        3      4     1     1      3        6   5   6   6  
1      5        3      3     1     1      3        4   5   5   6  
2      4        3      2     2     3      3       10   7   8  10  
3      3        2      2     1     1      5        2  15  14  15  
4      4        3      2     1     2      5        4   6  10  10  

[5 rows x 33 columns]

In [8]:
df_mat.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 395 entries, 0 to 394
Data columns (total 33 columns):
 #   Column      Non-Null Count  Dtype 
---  ------      --------------  ----- 
 0   school      395 non-null    object
 1   sex         395 non-null    object
 2   age         395 non-null    int64 
 3   address     395 non-null    object
 4   famsize     395 non-null    object
 5   Pstatus     395 non-null    object
 6   Medu        395 non-null    int64 
 7   Fedu        395 non-null    int64 
 8   Mjob        395 non-null    object
 9   Fjob        395 non-null    object
 10  reason      395 non-null    object
 11  guardian    395 non-null    object
 12  traveltime  395 non-null    int64 
 13  studytime   395 non-null    int64 
 14  failures    395 non-null    int64 
 15  schoolsup   395 non-null    object
 16  famsup      395 non-null    object
 17  paid        395 non-null    object
 18  activities  395 non-null    object
 19  nursery     395 non-null    object
 20  higher    

In [9]:
df_all = pd.concat([df_pt, df_mat])

In [10]:
df_all

school sex  age address famsize Pstatus  Medu  Fedu      Mjob      Fjob  \
0       GP   F   18       U     GT3       A     4     4   at_home   teacher   
1       GP   F   17       U     GT3       T     1     1   at_home     other   
2       GP   F   15       U     LE3       T     1     1   at_home     other   
3       GP   F   15       U     GT3       T     4     2    health  services   
4       GP   F   16       U     GT3       T     3     3     other     other   
..     ...  ..  ...     ...     ...     ...   ...   ...       ...       ...   
390     MS   M   20       U     LE3       A     2     2  services  services   
391     MS   M   17       U     LE3       T     3     1  services  services   
392     MS   M   21       R     GT3       T     1     1     other     other   
393     MS   M   18       R     LE3       T     3     2  services     other   
394     MS   M   19       U     LE3       T     1     1     other   at_home   

     ... famrel freetime  goout  Dalc  Walc health absences  G1  G2  G3  
0    ...      4        3      4     1     1      3        4   0  11  11  
1    ...      5        3      3     1     1      3        2   9  11  11  
2    ...      4        3      2     2     3      3        6  12  13  12  
3    ...      3        2      2     1     1      5        0  14  14  14  
4    ...      4        3      2     1     2      5        0  11  13  13  
..   ...    ...      ...    ...   ...   ...    ...      ...  ..  ..  ..  
390  ...      5        5      4     4     5      4       11   9   9   9  
391  ...      2        4      5     3     4      2        3  14  16  16  
392  ...      5        5      3     3     3      3        3  10   8   7  
393  ...      4        4      1     3     4      5        0  11  12  10  
394  ...      3        2      3     3     3      5        5   8   9   9  

[1044 rows x 33 columns]

In [11]:
df_all.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 1044 entries, 0 to 394
Data columns (total 33 columns):
 #   Column      Non-Null Count  Dtype 
---  ------      --------------  ----- 
 0   school      1044 non-null   object
 1   sex         1044 non-null   object
 2   age         1044 non-null   int64 
 3   address     1044 non-null   object
 4   famsize     1044 non-null   object
 5   Pstatus     1044 non-null   object
 6   Medu        1044 non-null   int64 
 7   Fedu        1044 non-null   int64 
 8   Mjob        1044 non-null   object
 9   Fjob        1044 non-null   object
 10  reason      1044 non-null   object
 11  guardian    1044 non-null   object
 12  traveltime  1044 non-null   int64 
 13  studytime   1044 non-null   int64 
 14  failures    1044 non-null   int64 
 15  schoolsup   1044 non-null   object
 16  famsup      1044 non-null   object
 17  paid        1044 non-null   object
 18  activities  1044 non-null   object
 19  nursery     1044 non-null   object
 20  higher   

## Qual é a diferente entre alunos que têm os pais **casados** e **separados** e que conseguiram pagar aulas de reforço e têm interesse em fazer um curso superior ?

Vamos utilizar as seguintes informações para o nosso estudo:
sex - sexo do aluno (binário: 'F' - feminino ou 'M' - masculino)

    1. age - idade do aluno (numérico: de 15 a 22)
    2. address – tipo de endereço residencial do aluno (binário: ‘U’ – urbano ou ‘R’ – rural)
    3. Pstatus - situação de coabitação dos pais (binário: 'T' - morando juntos ou 'A' - separados)
    4. Medu - escolaridade da mãe (numérico: 0 - nenhuma, 1 - ensino fundamental (4º ano), 2 - 5º ao 9º ano, 3 - ensino médio ou 4 - ensino superior)
    5. Fedu - escolaridade do pai (numérico: 0 - nenhuma, 1 - ensino fundamental (4º ano), 2 - 5º ao 9º ano, 3 - ensino médio ou 4 - ensino superior)
    6. Mjob - trabalho da mãe (nominal: 'professor', 'cuidados de saúde', 'serviços' civis (por exemplo, administrativo ou policial), 'em_casa' ou 'outros')
    7. Fjob - trabalho do pai (nominal: 'professor', 'cuidados de saúde', 'serviços' civis (por exemplo, administrativo ou policial), 'em_casa' ou 'outros')
    8. schoolup - apoio educativo extra (binário: sim ou não)
    9. paid - extras pagas dentro da disciplina do curso (Matemática ou Português) (binário: sim ou não)
    10. higher - quer cursar ensino superior (binário: sim ou não)
    11. internet - Acesso à internet em casa (binário: sim ou não)

Estas informações escolhidas de forma subjetiva.

Fazer um agrupamento dos dados por sexo e Pstatus.

In [12]:
df_filter= df_all[['sex','Pstatus', 'age', 'schoolsup', 'paid','higher', 'internet', 'Medu', 'Fedu', 'Mjob', 'Fjob', 'address', 'G1', 'G2', 'G3']]

In [13]:
df_group1 = df_filter.groupby(['sex', 'Pstatus'])

In [14]:
df_group1.count()

age  schoolsup  paid  higher  internet  Medu  Fedu  Mjob  Fjob  \
sex Pstatus                                                                   
F   A         77         77    77      77        77    77    77    77    77   
    T        514        514   514     514       514   514   514   514   514   
M   A         44         44    44      44        44    44    44    44    44   
    T        409        409   409     409       409   409   409   409   409   

             address   G1   G2   G3  
sex Pstatus                          
F   A             77   77   77   77  
    T            514  514  514  514  
M   A             44   44   44   44  
    T            409  409  409  409

Fazendo o agrupamento dos dados por Sexo e Situação de Coabitação dos Pais. Daí, ao analisar as carecteristicas selecionadas, tais como, a situação de coabitação dos pais, apoio educativo extra, se pagou aulas de reforço, a quantidade de alunos que têm interesse em fazer um curso superior é de **514** para as **mulheres** e **409** para os **homens**. Isso indica que a **base familiar** conta muito para a continuidade nos estudos de acordo com esses dados.

#### Filtrando os dados com o Pstatus igual a T (morando juntos)

In [15]:
df_filter_pstatus = df_filter[df_filter['Pstatus']=="T"]
df_filter_pstatus

sex Pstatus  age schoolsup paid higher internet  Medu  Fedu      Mjob  \
1     F       T   17        no   no    yes      yes     1     1   at_home   
2     F       T   15       yes   no    yes      yes     1     1   at_home   
3     F       T   15        no   no    yes      yes     4     2    health   
4     F       T   16        no   no    yes       no     3     3     other   
5     M       T   16        no   no    yes      yes     4     3  services   
..   ..     ...  ...       ...  ...    ...      ...   ...   ...       ...   
389   F       T   18        no   no    yes       no     1     1     other   
391   M       T   17        no   no    yes      yes     3     1  services   
392   M       T   21        no   no    yes       no     1     1     other   
393   M       T   18        no   no    yes      yes     3     2  services   
394   M       T   19        no   no    yes      yes     1     1     other   

         Fjob address  G1  G2  G3  
1       other       U   9  11  11  
2       other       U  12  13  12  
3    services       U  14  14  14  
4       other       U  11  13  13  
5       other       U  12  12  13  
..        ...     ...  ..  ..  ..  
389     other       U   6   5   0  
391  services       U  14  16  16  
392     other       R  10   8   7  
393     other       R  11  12  10  
394   at_home       U   8   9   9  

[923 rows x 15 columns]

Levando em consideração os dados de Sexo, schoolup (apoio educativo extra), paid (extras pagas dentro da disciplina do curso (Matemática ou Português)) e higher (quer cursar ensino superior).

In [16]:
df_filter_pstatus[['sex', 'schoolsup', 'paid', 'higher']].value_counts()

sex  schoolsup  paid  higher
F    no         no    yes       318
M    no         no    yes       254
F    no         yes   yes        95
M    no         yes   yes        78
F    yes        no    yes        56
M    no         no    no         46
F    no         no    no         26
M    yes        no    yes        20
F    yes        yes   yes        17
M    yes        yes   yes         7
     no         yes   no          2
     yes        no    no          2
F    no         yes   no          1
     yes        no    no          1
dtype: int64

Um dado interessante os alunos que não tiveram aulas extras são os que têm mais interesse em fazer um curso superior. A maioria fica para as **mulheres**, **318** e **254 para os homens**.

In [17]:
df_filter_pstatus[['sex', 'schoolsup', 'paid', 'higher', 'address']].value_counts()

sex  schoolsup  paid  higher  address
F    no         no    yes     U          222
M    no         no    yes     U          176
F    no         no    yes     R           96
M    no         no    yes     R           78
F    no         yes   yes     U           76
M    no         yes   yes     U           65
F    yes        no    yes     U           42
M    no         no    no      U           32
F    no         yes   yes     R           19
M    yes        no    yes     U           16
F    no         no    no      U           15
M    no         no    no      R           14
F    yes        no    yes     R           14
M    no         yes   yes     R           13
F    no         no    no      R           11
     yes        yes   yes     U           10
                              R            7
M    yes        no    yes     R            4
                yes   yes     U            4
                              R            3
     no         yes   no      R            2
     yes        n

E quando a informação de Endereço (zona rural ou urbana), nota-se que para ambos sexo, homem ou mulher, a maioria dos alunos que tem interesse em fazer um curso superior é dos alunos que moram na cidade, **222** para o sexo **feminino** e **176** para o sexo **masculino**.

Agrupar por sexo.

In [18]:
df_filter_pstatus.groupby(by='sex')['G1', 'G2','G3'].mean()

<ipython-input-18-fac61fa80196>:1: FutureWarning: Indexing with multiple keys (implicitly converted to a tuple of keys) will be deprecated, use a list instead.
  df_filter_pstatus.groupby(by='sex')['G1', 'G2','G3'].mean()


G1         G2         G3
sex                                 
F    11.408560  11.464981  11.554475
M    10.968215  10.943765  10.977995

Como resultado, vimos que a nota média das mulheres é superior a dos homens em Português e Matemática. G1, nota do primeiro período para as mulheres, 11.408560 e 10.968215, já para o segundo período é de 11.464981 e 10.943765, masculino e feminino, respectivamente. Por fim, a média das notas é de 11.554475 e 10.977995, feminino e masculino.

#### Vamos analisar os dados dos alunos que têm os pais separados pois vimos que essa era a informação com maior diferença nos dados.

In [19]:
df_fil_not_t = df_filter[df_filter['Pstatus']=="A"]
df_fil_not_t

sex Pstatus  age schoolsup paid higher internet  Medu  Fedu      Mjob  \
0     F       A   18       yes   no    yes       no     4     4   at_home   
7     F       A   17       yes   no    yes       no     4     4     other   
8     M       A   15        no   no    yes      yes     3     2  services   
14    M       A   15        no   no    yes      yes     2     2     other   
28    M       A   16       yes  yes    yes      yes     3     4  services   
..   ..     ...  ...       ...  ...    ...      ...   ...   ...       ...   
339   F       A   17        no  yes    yes      yes     3     2     other   
343   F       A   17        no   no    yes      yes     2     2   at_home   
357   F       A   17        no   no    yes       no     3     2  services   
360   F       A   18        no   no    yes       no     1     4   at_home   
390   M       A   20        no  yes    yes       no     2     2  services   

         Fjob address  G1  G2  G3  
0     teacher       U   0  11  11  
7     teacher       U  10  13  13  
8       other       U  15  16  17  
14      other       U  14  14  15  
28      other       U  12  12  13  
..        ...     ...  ..  ..  ..  
339     other       R   9  10  10  
343   at_home       U   9   8   0  
357     other       U  12  12  11  
360     other       R  13  13  13  
390  services       U   9   9   9  

[121 rows x 15 columns]

In [20]:
df_fil_not_t[["sex", "schoolsup", "paid", "higher"]].value_counts()

sex  schoolsup  paid  higher
F    no         no    yes       45
M    no         no    yes       34
F    no         no    no        10
     yes        no    yes       10
     no         yes   yes        8
M    no         yes   yes        7
F    yes        yes   yes        3
M    yes        no    yes        2
F    no         yes   no         1
M    yes        yes   yes        1
dtype: int64

Similar com o que aconteceu para os alunos que tem os paises vivendo na mesma casa, os alunos que não teve aulas extras são aqueles que têm mais interesse em fazer um curso superior. Vamos analisar se o fator endereço também é um fator determinante para os alunos.

In [21]:
df_fil_not_t[["sex", "schoolsup", "paid", "higher", 'address']].value_counts()

sex  schoolsup  paid  higher  address
F    no         no    yes     U          37
M    no         no    yes     U          29
F    yes        no    yes     U          10
     no         no    yes     R           8
                      no      U           6
                yes   yes     U           6
M    no         no    yes     R           5
                yes   yes     U           5
F    no         no    no      R           4
     yes        yes   yes     U           3
     no         yes   yes     R           2
M    no         yes   yes     R           2
     yes        no    yes     U           2
F    no         yes   no      R           1
M    yes        yes   yes     U           1
dtype: int64

Assim como foi na análise anterior, os alunos que moram na cidade tem mais interesse dar continuidade nos estudos. Isso nós leva a entender, que, para aumentar a quantidade de alunos na faculdade, é investir em alunos que moram em zona rural para que o interesse dos alunos aumentem ou o acesso à educação também.

In [22]:
df_fil_not_t.groupby(by="sex")["G1", "G2", "G3"].mean()

<ipython-input-22-68670c33e766>:1: FutureWarning: Indexing with multiple keys (implicitly converted to a tuple of keys) will be deprecated, use a list instead.
  df_fil_not_t.groupby(by="sex")["G1", "G2", "G3"].mean()


G1         G2         G3
sex                                 
F    10.415584  10.337662  10.740260
M    12.613636  13.090909  13.295455

Neste caso, para os alunos que moram em zona rural, a média geral das notas de Português e Matemática, é maior para os homens, **13.29**, enquanto das mulheres que moram na cidade é de **11.55**. O que pode contar nesse caso, pode ser o dado **romantic** (com relacionamento amoroso) na qual pode informar que as mulheres que moram na zona rural casam cedo e precisam dar mais atenação a casa e filhos.

In [23]:
df_all.loc[(df_all['romantic']=='yes') & (df_all['address']=='R')].groupby(by=['romantic', 'sex'])['romantic', 'sex','G1', 'G2', 'G3'].mean()

<ipython-input-23-c33fbb95f5dd>:1: FutureWarning: Indexing with multiple keys (implicitly converted to a tuple of keys) will be deprecated, use a list instead.
  df_all.loc[(df_all['romantic']=='yes') & (df_all['address']=='R')].groupby(by=['romantic', 'sex'])['romantic', 'sex','G1', 'G2', 'G3'].mean()
<ipython-input-23-c33fbb95f5dd>:1: FutureWarning: The default value of numeric_only in DataFrameGroupBy.mean is deprecated. In a future version, numeric_only will default to False. Either specify numeric_only or select only columns which should be valid for the function.
  df_all.loc[(df_all['romantic']=='yes') & (df_all['address']=='R')].groupby(by=['romantic', 'sex'])['romantic', 'sex','G1', 'G2', 'G3'].mean()


G1         G2         G3
romantic sex                                 
yes      F    10.266667   9.750000   9.633333
         M    10.260870  10.217391  10.239130

Conforme apresentado nos dados, o fator endereço e relacionamento amoroso impactam para as mulheres.

## Resultado

Foram analisando alguns dos dados sobre uma pesquisa das notas de Português e Matemática de alguns alunos extraídos deste conjunto de dados [link](https://colab.research.google.com/corgiredirector?site=https%3A%2F%2Farchive.ics.uci.edu%2Fdataset%2F320%2Fstudent%2Bperformance). À partir desses dados, conseguimos descobrir que as notas das mulheres são maiores quando os alunos moram na cidade e que ter ou não aulas particulares motiva os alunos a fazerem curso superior.

Como próximos passo, aconselho avançar nos estudos e apresentar esses dados através de gráficos utilizando o Plotly.

In [24]:
import pandas as pd
import seaborn as sns
from matplotlib import pyplot as plt

In [25]:
%matplotlib inline

In [26]:
df_all.info()


<class 'pandas.core.frame.DataFrame'>
Int64Index: 1044 entries, 0 to 394
Data columns (total 33 columns):
 #   Column      Non-Null Count  Dtype 
---  ------      --------------  ----- 
 0   school      1044 non-null   object
 1   sex         1044 non-null   object
 2   age         1044 non-null   int64 
 3   address     1044 non-null   object
 4   famsize     1044 non-null   object
 5   Pstatus     1044 non-null   object
 6   Medu        1044 non-null   int64 
 7   Fedu        1044 non-null   int64 
 8   Mjob        1044 non-null   object
 9   Fjob        1044 non-null   object
 10  reason      1044 non-null   object
 11  guardian    1044 non-null   object
 12  traveltime  1044 non-null   int64 
 13  studytime   1044 non-null   int64 
 14  failures    1044 non-null   int64 
 15  schoolsup   1044 non-null   object
 16  famsup      1044 non-null   object
 17  paid        1044 non-null   object
 18  activities  1044 non-null   object
 19  nursery     1044 non-null   object
 20  higher   

In [27]:
import pandas as pd
import plotly.express as px
from matplotlib import pyplot as plt

fig = px.bar(df_all, x='sex', y='G3', color='sex', title='Gráfico de Barras: Média Geral de Portugês e  Matemática por Sexo')
fig.show()
